### PR TITLE
Run E2E test with the current nuts-node image

### PR DESCRIPTION
**Pull request by dmeijboom**
This PR implements running the E2E tests for the actual current image build from the last commit in the PR.

Flow:
1. Docker build from the last commit
2. Docker push to GitHub Container Registry
3. Start Earthly with E2E tests
4. Earthly pulls docker image
5. E2E tests are running and all is well

### CODE REVIEW

**Comment by dmeijboom** [comment]
Note that the E2E tests requires: https://github.com/nuts-foundation/nuts-go-e2e-test/pull/8 to finish successfully.

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,24 +47,27 @@ jobs:
...
-          command: |
-            git clone https://github.com/nuts-foundation/nuts-go-e2e-test.git && \
-              cd nuts-go-e2e-test && \
-              find . -type f -name "docker-compose.yml" | xargs -I{} sed "s/nutsfoundation\/nuts-node:master/nuts-node-ci\/${CIRCLE_SHA1}/g" {}
-      - run:
-          name: Run E2E tests
-          command: |
-            cd nuts-go-e2e-test && \
-              docker run --privileged -t -v /var/run/docker.sock:/var/run/docker.sock -v $(pwd):/workspace -v earthly-tmp:/tmp/earthly:rw earthly/earthly:latest -P +all
+#  e2e-tests:
```
Comment (woutslakhorst) [inline .circleci/config.yml:50]: just remove the commented stuff. Will be in Github history if you need it.

**Review by woutslakhorst — COMMENTED** [review]
Rerun test after e2e test fix and then if green 👍

File: .circleci/config.yml (hunk 1 above)
Comment (dmeijboom) [reply .circleci/config.yml]: Oops, removed!

**Comment by dmeijboom** [comment]
> Rerun test after e2e test fix and then if green 👍

Done 😃

**Comment by woutslakhorst** [comment]
So basic fix for deleting packages is:

run the delete action from the master branch, running it from a PR/fork doesn't allow for deleting. Also use the head_sha and not github.sha since the latter is the sha resulting from a pull merge commit.

On the master branch, I added a on: workflow_run action that is triggered by completed e2e test actions. It'll then delete the package from the context of the master branch....

**Review by woutslakhorst — DISMISSED** [review]

**Comment by woutslakhorst** [comment]
I spoke too soon.
Most current github actions use the graphql API to delete a package. This is not possible for containers.
The github REST api should be used.

A call to api.github.com/orgs/nuts-foundation/packages/container/nuts-node-ci/versions

yields:

```
[
    {
        "id": 4576075,
        "name": "sha256:8500f52154ced49185aed571153fdf679cbeee8d40a0c70918cc73f55af5b833",
        "url": "https://api.github.com/orgs/nuts-foundation/packages/container/nuts-node-ci/versions/4576075",
        "package_html_url": "https://github.com/orgs/nuts-foundation/packages/container/package/nuts-node-ci",
        "created_at": "2021-07-23T14:05:51Z",
        "updated_at": "2021-07-23T14:05:51Z",
        "html_url": "https://github.com/orgs/nuts-foundation/packages/container/nuts-node-ci/4576075",
        "metadata": {
            "package_type": "container",
            "container": {
                "tags": [
                    "6dfdedd749881c4549a12d4808b06d6dfea4bd2b"
                ]
            }
        }
    }
]
```

Version then have to be deleted by `id`, where matching needs to take place on the `metadata.container.tags`

**Review by woutslakhorst — APPROVED** [review]
The missing link was a personal access token with delete rights. The ghcr-delete-image-action from bots-house worked...

**Pull request merged**